import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/AMPStaticProcessor.java

```diff
@@ -312,27 +314,51 @@ private void updateProject(JsonBean project, InternalDocument source, List<Field
 		}
 		// Process transactions
 		if (hasTransactions) {
-			JsonBean fundings = getTransactions(source, fieldMappings, valueMappings);
+			List<JsonBean> fundings = null;
+			switch(importRequest.getImportOption()) {
+			   case OVERWRITE_ALL_FUNDING:
+				   fundings = new ArrayList<>();
+				   fundings.add(getSourceFundings(source, fieldMappings, valueMappings));
+				   break;
+			   case ONLY_ADD_NEW_FUNDING:
+				   fundings = addNewFunding(source, fieldMappings, valueMappings, project);
+				   break;
+			   case REPLACE_DONOR_FUNDING:
+				   fundings = replaceDonorTransactions(source, fieldMappings, valueMappings, project);
+				   break;
+			   default:
+					break;
+			}
+			
 			if (fundings != null) {
-				project.set("fundings", fundings);
-				Optional<Field> org = fieldList.stream().filter(n -> {
-					return n.getFieldName().equals("donor_organization");
-				}).findFirst();
-				if (org.isPresent()) {
-					List<JsonBean> listDonorOrganizations = new ArrayList<JsonBean>();
-					JsonBean donorRole = new JsonBean();
-					donorRole.set("organization", fundings.get("donor_organization_id"));
-					donorRole.set("role", 1);
-					if (org.get().isPercentage()) {
-						donorRole.set("percentage", 100);
-					}
-					listDonorOrganizations.add(donorRole);
-					project.set("donor_organization", listDonorOrganizations);
-				}
+				project.set("fundings", fundings);				
+				project.set("donor_organization", getDonorOrgs(fundings));
 			}
+			
 		}
 	}
 
+	private List<JsonBean> getDonorOrgs(List<JsonBean> fundings){
+		List<JsonBean> listDonorOrganizations = new ArrayList<JsonBean>();
+		for(JsonBean funding : fundings) {
+			Optional<Field> org = fieldList.stream().filter(n -> {
+				return n.getFieldName().equals("donor_organization");
+			}).findFirst();
+			if (org.isPresent()) {
+				
+				JsonBean donorRole = new JsonBean();
+				donorRole.set("organization", funding.get("donor_organization_id"));
+				donorRole.set("role", 1);
+				if (org.get().isPercentage()) {
+					donorRole.set("percentage", 100);
```

Comment:
what if we have more donors? We have spit the percentages

Reply:
@vchihai - I have removed the code for setting percentage for donors